x-pack/platform/plugins/shared/task_manager/server/lib/get_first_run_at.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RRule } from '@kbn/rrule';
+import type { Logger } from '@kbn/core/server';
+import type { RruleSchedule, TaskInstance } from '../task';
+
+export function getFirstRunAt({
+  taskInstance,
+  logger,
+}: {
+  taskInstance: TaskInstance;
+  logger: Logger;
+}): string {
+  if (taskInstance.runAt) {
+    return taskInstance.runAt.toISOString();
+  }
+  const now = new Date();
+  const nowString = now.toISOString();
+
+  if (taskInstance.schedule?.rrule && rruleHasFixedTime(taskInstance.schedule.rrule)) {
+    try {
+      const rrule = new RRule({
+        ...taskInstance.schedule.rrule,
+        bysecond: [0],
```

Comment:
why do we need the `bysecond` here and not in the `get_next_run` function below?

wondering if we could create a function for handling calculating the next runAt with rrule that can be used here and in the `get_next_run` calculation

Reply:
After our discussion we wanted the seconds always to be '0' for the configs that has fixed time.
A config like `freq:daily`, `byhour:13` will be = every day at `13:00:00`
or `freq:hourly`,`byminute:15` will be = every hour at 15:00.

get_next_run uses the existing runAt as an input to `dtstart`, so it picks up the initial config. no need to set it there again.

and for the configs without fixed time rrule picks up the current time's second.